Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -393,6 +398,23 @@ void WiiSocket::Update(bool read, bool write, bool except)
           {
           case IOCTLV_NET_SSL_DOHANDSHAKE:
           {
+            // The Wii allows a socket with an in-progress connection to
+            // perform the SSL handshake. MbedTLS doesn't support it so
+            // we have to check it manually.
+            const ConnectingState state = GetConnectingState();
```

Comment:
If you decide to incorporate my suggestions, this could be:

`connecting_state = GetConnectingState()`

Reply:
I'll have a look at it. Since I'm in the `WiiSocket` class, that seems doable, indeed.